### PR TITLE
Ci/use z3 image

### PR DESCRIPTION
**Pull request by F-WRunTime**
Updating Kontrol CI to use prebuild docker image of z3. 
Updating Published Kontrol iamge to use same version of Z3 Kontrol is built with

Foundry is setup inconsisntly between published Kontrol image and the CI image. Should this also be address see below: 

From [.github/workflows/Dockerfile,](https://github.com/runtimeverification/kontrol/blob/master/.github/workflows/Dockerfile#L6C1-L7C1) version is pinned to a nightly build: 
```
FROM ghcr.io/foundry-rs/foundry:nightly-ca67d15f4abd46394b324c50e21e66f306a1162d as FOUNDRY
```
From [Dockerfile ](https://github.com/runtimeverification/kontrol/blob/master/Dockerfile#L28)(top of repository) the published kontrol image always pulls the latest nightly Foundry version... This could cause inconsistent results? 
```
ENV PATH=/home/user/.foundry/bin:${PATH}
RUN    curl -L https://foundry.paradigm.xyz | bash \
    && foundryup
```

### CODE REVIEW

**Comment by F-WRunTime** [comment]
This is addressing https://github.com/runtimeverification/devops/issues/165

**Comment by F-WRunTime** [comment]
Auto updates in CI [here ](https://github.com/runtimeverification/devops/pull/166)

File: deps/k_release [hunk 1]
```diff
@@ -1 +1 @@
-6.0.87
+6.0.87
```
Comment (tothtamas28) [inline deps/k_release:1]: Consider adding a newline.

File: deps/z3 [hunk 2]
```diff
@@ -0,0 +1 @@
+4.12.1
```
Comment (tothtamas28) [inline deps/z3:1]: Consider adding a newline.

File: .github/workflows/Dockerfile [hunk 3]
```diff
@@ -6,7 +6,7 @@ ARG LLVM_VERSION
 FROM ghcr.io/foundry-rs/foundry:nightly-ca67d15f4abd46394b324c50e21e66f306a1162d as FOUNDRY
 
 ARG Z3_VERSION
-FROM z3:${Z3_VERSION} as Z3
+FROM runtimeverificationinc/ubuntu-jammy-z3:${Z3_VERSION} as Z3
```
Comment (tothtamas28) [inline .github/workflows/Dockerfile:9]: Is the additional

```Dockerfile
COPY --from=Z3 /usr/bin/z3 /usr/bin/z3
```

not necessary here?

**Review by tothtamas28 — APPROVED** [review]

File: deps/k_release (hunk 1 above)
Comment (tothtamas28) [reply deps/k_release]: Sorry for the ambiguous wording, what I meant is consider changing `'6.0.87'` to `'6.0.87\n'`.

**Comment by F-WRunTime** [comment]
Keep getting this error on each run: 
```
[Info#booster] Simplifying a pattern
[Warn] user error (Unexpected result from the SMT solver:
  Command: (push 1 )
  Expected: success
  Result: (error "line 3624 column 8: canceled" )
)
``` 
Is this a known issue?

**Comment by tothtamas28** [comment]
> Is this a known issue?

Yes, it's an SMT solver timeout. Proof should hopefully pass after a restart.

**Pull request merged**